docs/src/pages/system/advanced/advanced.md

```diff
@@ -0,0 +1,21 @@
+# Advanced
+
+<p class="description">Here you can find examples of how you can use the system on your custom components.</p>
+
+## Adding the `sx` prop on your custom components
+
+The `unstable_styleFunctionSx` utility adds the support for the `sx` in your own components.
+
+{{"demo": "pages/system/advanced/StyleFunctionSxDemo.js"}}
+
+### Built-in `sx` from the core
```

Comment:
The advantage is that you can add it to more complex custom components, and it would behave is it does on the core MUI components. 

I had this initially as first secrtion, as it seemd like the easiest way to add it, but @oliviertassinari had a valid point that this is a page for the system and the first thing we are advertising is something from the core, that’s why we moved it as second section. I am ok with any order, but I think I should probably rephrase to explain why it is valuable (if it is not obvious to you guys, than this is a bad docs from my side :/)

Reply:
> you can add it to more complex custom components

I think the point is that in this instance (already using core) you could just use Box as your root component.

> @oliviertassinari had a valid point that this is a page for the system and the first thing we are advertising is something from the core

Notwithstanding the above, it seems to be about how to use the styles system without using Box.

Reply:
Let's rewind a bit, there are multiple layers, by level of abstractions:

1. the system functions
2. the styled engine
3. the styled() implementation
4. the theme
5. the Box
6. the usage of `sx`

This echo to a discussion we had about moving Box into the system.

Reply:
> the styled engine
"style engine". The sooner we rename that the better.

That aside, having rewound (and played forward) what was your point?

Reply:
> @eps1lon - don't want to put words in your mouth here...

That is my point. I'm just completely lost why I would use this functionality. The issues described are not voiced by any users (how could they if they didn't have time to use `sx`) and the API itself hasn't had any time to incubate. But we're already pushing everything we can to production level without having documentation. The pages added just showcase usage which is trivial.

In my opinion our internal styled should not be published at this point. We didn't have any time to battle-test the API.

But I think I've already said this enough so there's no point repeating anymore. Just said it because I was pinged explicitly.

Reply:
Should we then hold off with these changes until we have an actual requirement for something like this?

Reply:
I am either way describing experimental and unstable API so maybe we should just wait and see the feedback on the prop first...

Reply:
There is one aspect I would like to raise: listening to users has limits too. No customer [ever asked](https://www.sec.gov/Archives/edgar/data/1018724/000119312517120198/d373368dex991.htm) Amazon to build Amazon prime, they have done it anyway. I think that what's fundamentally important is either or not can we envision this abstraction to be used or not? In our case, more by front-end developers or designers (we have a large audience of full-stack developers, so it doesn't help).

Reply:
As a compromise, how about removing this section ("Built-in `sx` from the core") for now? Sure, it's another way of achieving the same end result, but it doesn't seem to add any value over just using Box. Presenting multiple ways to achieve the same thing just leads to confusion.

In the other two sections, we emphasise the benefits of each approach. (Use without core / less overhead, respectively.)

Reply:
Removed that section and added reasons of why someone may use these utilities. I hope is much clear now :)